### ISSUE TITLE
Unclear priority while handling routes

### ISSUE THREAD

**Issue by ghost** [actix/actix-web#75]
Why does

```
HttpServer::new(
        move || {
            Application::new()
                .handler("{file_name}.{ext}", fs::StaticFiles::new("./assets", false))
                .resource("{language_code}", move |r| r.h(serve))
                .default_resource({ println!("default"); |r| r.method(Method::GET).h(NormalizePath::new(
                    false, true, StatusCode::MOVED_PERMANENTLY))})
        })
        .bind(url).unwrap()
        .start();
```
always match on the second rule?

I would expect: `localhost:8000/en.test` to be matched by the first handler, but this isn't the case, it's the second one that gets invoked.

Also, the third one is never invoked, like e.g. `http://localhost:8000/test/1234` renders a white page.

Bonus question:
Can I somehow combine the second with the third?
Like I would like to remove all trailing slashes, always... no matter how many of them there are. Is there anything for that already, or should I do that manually?

**Comment by fafhrd91** [comment]
1. `.handler()` has lower priority, application first matches resources and then handler.
also `.handler()` use prefix, it does not work with regex.

2. to remove trailing slashes you can use https://actix.github.io/actix-web/actix_web/struct.NormalizePath.html

**Comment by ghost** [comment]
Thanks, I already am using that, but it never gets hit.

**Comment by ghost** [comment]
Tried to fix the first issue with another resource: 

`.resource("{file_name}.{ext}", |r| r.method(Method::GET).f(assets))`

```
fn assets(req: HttpRequest) -> actix_web::Result<fs::NamedFile> {
    let file_name = req.match_info().query::<String>("file_name")?;
    let file_ext = req.match_info().query::<String>("ext")?;

    println!("filename: {:?}", file_name);
    println!("file_ext: {:?}", file_ext);

    // FIXME
    let path = format!("./assets/{:?}.{:?}", file_name, file_ext);

    Ok(fs::NamedFile::open(path)?)
}
```

Now that one gets called (thanks for the tip with the handler and resource priority!)

But the regex seems to be off. I mean: `http://localhost:8000/test.js` gives:

`filename: "test."
file_ext: "s"
`

**Comment by fafhrd91** [comment]
it is bug, i am fixing it

**Comment by fafhrd91** [comment]
NormalizePath is fixed by ddc8239

**Comment by fafhrd91** [comment]
`"./assets/{:?}.{:?}"` pattern should be fixed by 4d81186
